menoh/mkldnn_with_generic_fallback/backend/mkldnn/mkldnn_context.cpp

```diff
@@ -37,86 +39,98 @@ namespace menoh_impl {
-                                               "varibale table: "
-                                            << context_pair.first << std::endl;
-                                    procedure copy_proc; // copy from other
-                                                         // context to array
-                                    array arr;
-                                    std::tie(copy_proc, arr) = *found;
-                                    assert(arr.dims().size() == 2 ||
-                                           arr.dims().size() == 4);
-                                    new_copy_procedure_list.push_back(
-                                      copy_proc);
-                                    input_list.push_back(arr);
-                                    is_found_from_other_context = true;
+                                            << " is found in common input table"
+                                            << std::endl;
                                     break;
                                 }
-                            }
-                            assert(is_found_from_other_context);
-                        } while(false);
-                    }
 
-                    // make primitives and
-                    std::vector<mkldnn::primitive> new_primitive_list;
-                    std::vector<std::pair<std::string, mkldnn::memory>>
-                      new_output_memory_list;
-                    std::vector<mkldnn::memory> new_temp_memory_list;
-                    try {
+                                // search in other contexts' variable table
+                                bool is_found_from_other_context = false;
+                                for(auto const& context_pair : context_list) {
+                                    if(context_pair.first == context_name) {
+                                        continue; // skip self
+                                    }
+                                    auto found =
+                                      context_pair.second->try_to_get_variable(
+                                        input_name);
+                                    if(found) {
+                                        *logger
+                                          << input_name
+                                          << " is found in other context's "
+                                             "varibale table: "
+                                          << context_pair.first << std::endl;
+                                        procedure copy_proc; // copy from other
+                                                             // context to array
+                                        array arr;
+                                        std::tie(copy_proc, arr) = *found;
+                                        assert(arr.dims().size() == 2 ||
+                                               arr.dims().size() == 4);
```

Comment:
Who is responsible to ensure this condition?
If it is a user who input ONNX file, then it would be better to throw exception rather than just using `assert` (which will be eliminated in the Relase build).

Also, note that an exception is thrown in the following place:
https://github.com/pfnet-research/menoh/blob/476ec3a7b133eda9bf7b030ba47577eddb9ed237/menoh/mkldnn_with_generic_fallback/backend/mkldnn/mkldnn_context.cpp#L67-L84